packages/addons/script/steamlink-rpi/source/default.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2019-present Team LibreELEC (https://libreelec.tv)
+
+import os
+import tarfile
+import subprocess
+import xbmcaddon
+import xbmcgui
+from hashlib import sha256
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from sys import exit
+from urllib.request import urlretrieve
+
+
+STEAMLINK_VERSION = "@STEAMLINK_VERSION@"
+STEAMLINK_HASH = "@STEAMLINK_HASH@"
+STEAMLINK_TARBALL_NAME = f"steamlink-rpi3-{STEAMLINK_VERSION}.tar.gz"
+STEAMLINK_URL = f"http://media.steampowered.com/steamlink/rpi/{STEAMLINK_TARBALL_NAME}"
+ADDON_DIR = xbmcaddon.Addon().getAddonInfo("path") + "/"
+
+
+def GetSHA256Hash(file_name):
+  """ Get sha256sum of file_name in 8kb chunks """
+  with open(file_name,"rb") as file:
+    SHA256HASH = sha256()
+    while True:
+      data_block = file.read(8192)
+      if not data_block:
+        break
+      SHA256HASH.update(data_block)
+  return SHA256HASH.hexdigest()
+
+def GetRPiProcessor():
+  """ Use vcgencmd to obtain cpu identifier as int """
+  VC_CMD_OUTPUT=subprocess.check_output(["vcgencmd", "otp_dump"], encoding="utf-8")
+
+  for line in VC_CMD_OUTPUT.splitlines():
+    if line[0:3] == "30:":
+      PROCESSOR=line.split(":")[1] # entire processor id
+      PROCESSOR=PROCESSOR[4:5]     # only cpu id
+      break
+  return int(PROCESSOR)
```

Comment:
Need to add `PROCESSOR=0` before the loop so that the `PROCESSOR` variable is known should `30:` not be processed within the loop (otherwise the `return` statement will throw an exception).